openml/runs/run.py

```diff
@@ -435,7 +449,7 @@ def _create_description_xml(self):
         # tags = run_environment + [well_formatted_time] + ['run_task'] + \
         #     [self.model.__module__ + "." + self.model.__class__.__name__]
         description = _to_dict(taskid=self.task_id, flow_id=self.flow_id,
-                               setup_string=_create_setup_string(self.model),
+                               setup_string=self.setup_string,
```

Comment:
before you used the function to create a setup string on the fly, however now you rely on the fact that there is already a setup string and that it's up to date. Are you sure this is intended? Since the function is rather fast, I would make a function call nonetheless (to keep the function easily readible)

Reply:
Retrieving the setup string depends on the model. I moved this to the creation of the run object (after actually running the model) where we can be sure that the model is available. Therefore, I think it's correct.